src/MixedMatrix.hpp

```diff
@@ -304,7 +350,7 @@ class MixedMatrix
 
     void GenerateRowStarts();
 
-    std::unique_ptr<mfem::SparseMatrix> M_;
+    mutable std::unique_ptr<mfem::SparseMatrix> M_;
```

Comment:
This is a good point. I think I make `M_` mutable mainly because of `MixedMatrix::getWeight() const`. This function should really just return a reference to `*M_`. Currently `M_` is changed in  getWeight() only if it is `nullptr` (similar to `get_pM()` and `get_pD()`). In the case when `M_` is rescaled, it is updated in `UpdateM()`, so `getWeight()` would not change `M_` because it is not `nullptr`. So "technically" the mass matrix is not changed in either case. But I agree this should be better designed.